lib/Doctrine/Inflector/Rules/English/Uninflected.php

```diff
@@ -31,6 +31,7 @@ public static function getPlural() : iterable
         yield new Pattern('middleware');
         yield new Pattern('people');
         yield new Pattern('software');
+        yield new Pattern('hardware');
```

Comment:
I wonder: Does every word ending with ware has the same pluralization? We could cover all those words with a regex pattern.

Reply:
I do believe so, except for the word "ware" itself which pluralizes to "wares".

Reply:
This can be covered with a regex.

Reply:
As long as we're dealing with nouns that seems to be case. As ware itself is a mass noun, and mass nouns by definition indicate a collection of multiple things. Hence, they have no separate plural form. See: https://en.oxforddictionaries.com/definition/ware#h47396550536000

The only edge case I could think of where a pattern could become tricky is when you go beyond nouns, and get names ending in -aware. Though such words would mostly be used as an adjective, rather than a noun.

@SenseException: Given these observations, would you like me to create a regex version instead?

Reply:
Interesting, so it is unclear whether "ware" is being used as a mass noun or a singular noun from code's perspective -- so should we then pluralize it or not?

Reply:
What would you think of the following regex?
`[A-Za-z]+ware$`

It excludes `ware` as such, allowing for the pluralisation to `wares` (which is mentioned in the Oxford dictionary example 1.2 of ware^1, but it would would exclude terms such as `hardware` or `software` from being pluralized as `hardwares` or `softwares`...